### PR TITLE
Tidied up some code using optional map

### PR DESCRIPTION
**Pull request by BenMaer**
Simply shortens/tidies up a few places of code, each of which were 3 lines, but could be written with just one using Optional's map function.

This feels like too small of change to mention in the changelog. If you disagree, let me know and I can make a mention.

I was also thinking of further condensing this code to the existing extension on `Disposables` in `AnonymousDisposable.swift`:
```
extension Disposables {
.....
    /// Constructs a new disposable with the given action used for disposal. Provides an optional `dispose` param for convenience
    ///
    /// - parameter dispose: Optional disposal action which would be run upon calling `dispose`.
    public static func create(with dispose: (() -> Void)?) -> Disposable {
        return dispose.map({ create(with: $0) }) ?? create()
    }

}
```

That way, the three places I cleaned up wouldn't each be repeating the same logic. Let me know if that's something you'd want.

### CODE REVIEW

File: RxSwift/Traits/PrimitiveSequence/Completable.swift [hunk 1]
```diff
@@ -123,12 +123,7 @@ extension PrimitiveSequenceType where Trait == CompletableTrait, Element == Swif
                 let callStack = [String]()
         #endif
 
-        let disposable: Disposable
-        if let onDisposed = onDisposed {
-            disposable = Disposables.create(with: onDisposed)
-        } else {
-            disposable = Disposables.create()
-        }
+        let disposable: Disposable = onDisposed.map({ Disposables.create(with: $0) }) ?? Disposables.create()
```
Comment (freak4pc) [inline RxSwift/Traits/PrimitiveSequence/Completable.swift:126]:
```suggestion
        let disposable: Disposable = onDisposed.map(Disposables.create(with:)) ?? Disposables.create()
```

File: RxSwift/Traits/PrimitiveSequence/Maybe.swift [hunk 2]
```diff
@@ -138,12 +138,7 @@ extension PrimitiveSequenceType where Trait == MaybeTrait {
         #else
             let callStack = [String]()
         #endif
-        let disposable: Disposable
-        if let onDisposed = onDisposed {
-            disposable = Disposables.create(with: onDisposed)
-        } else {
-            disposable = Disposables.create()
-        }
+        let disposable: Disposable = onDisposed.map({ Disposables.create(with: $0) }) ?? Disposables.create()
```
Comment (freak4pc) [inline RxSwift/Traits/PrimitiveSequence/Maybe.swift:141]: ditto

File: RxSwift/Traits/PrimitiveSequence/Single.swift [hunk 3]
```diff
@@ -135,12 +135,7 @@ extension PrimitiveSequenceType where Trait == SingleTrait {
             let callStack = [String]()
         #endif
 
-        let disposable: Disposable
-        if let onDisposed = onDisposed {
-            disposable = Disposables.create(with: onDisposed)
-        } else {
-            disposable = Disposables.create()
-        }
+        let disposable: Disposable = onDisposed.map({ Disposables.create(with: $0) }) ?? Disposables.create()
```
Comment (freak4pc) [inline RxSwift/Traits/PrimitiveSequence/Single.swift:138]: ditto :)

**Review by freak4pc — COMMENTED** [review]
Nice fix! I added a clean-up suggestion.

File: RxSwift/Traits/PrimitiveSequence/Completable.swift (hunk 1 above)
Comment (BenMaer) [reply RxSwift/Traits/PrimitiveSequence/Completable.swift]: Clever suggestion! I committed your suggestion, and will commit the others as well.

File: RxSwift/Traits/PrimitiveSequence/Maybe.swift (hunk 2 above)
Comment (BenMaer) [reply RxSwift/Traits/PrimitiveSequence/Maybe.swift]: Updated

File: RxSwift/Traits/PrimitiveSequence/Single.swift (hunk 3 above)
Comment (BenMaer) [reply RxSwift/Traits/PrimitiveSequence/Single.swift]: Updated

**Comment by BenMaer** [comment]
Hey @freak4pc , I made the updates you suggested. Clever call with that!

Let me know if there's anything else I can do.

**Review by freak4pc — APPROVED** [review]
Thanks!

**Pull request merged**